gr-iio/lib/ad9081_common_impl.cc

```diff
@@ -0,0 +1,106 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "ad9081_common_impl.h"
+
+#include <gnuradio/iio/ad9081_common.h>
+
+#include <cmath>
+#include <cstdint>
+
+namespace gr {
+namespace iio {
+
+long long phase_fold(float x)
```

Comment:
Unfortunately IIOs `iio_channel_attr_write_longlong()`, requires a `long long`s, because that's the only type guaranteed to have 64-bit on any system. If i'd use `int64_t` here i'd have to cast that to `long long` later on because on AMD64 `int64_t` is just `long`. This is really just a helper method for that one place, which is why i'm directly returning `long long`, though i suppose it could be moved to the header as `constexpr` / `inline`.

Reply:
Well, int64_t is a type that's guaranteed to be 64 bits on any system :) 
As said further down, this is in the end an API decision up to you, but if you have any public-facing API (i.e. what ends up as symbol in a shared library), and you *mean* 64 bit, please use an explicit 64 bit type. These casts, they don't compile to anything.